enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/upstream/UpstreamDatabaseStrategySelector.java

```diff
@@ -59,30 +61,28 @@ public UpstreamDatabaseStrategySelector( UpstreamDatabaseSelectionStrategy defau
 
     public MemberId bestUpstreamDatabase() throws UpstreamDatabaseSelectionException
     {
-        MemberId result = null;
-        for ( UpstreamDatabaseSelectionStrategy strategy : strategies )
+        MemberId result = strategies.stream()
```

Comment:
Has the logic changed here? I am not a fan of changing code if not needed and the previous code looks easy to read to me.

Reply:
The logic hasn't changed, however with @hugofirth we thought that streams were preferable to loops. I will keep it as is, if that is ok with you?

Reply:
The issue introduced above with the UpstreamSelectionExcpetion is a reason for why I would not change working code if there is no other reason than "style" which I find very subjective.

However this is my subjective opinion so I'll leave it up to you as long as the above issue is fixed :)

Reply:
My preference for the streaming style in this case had more to do with the fact that I felt `findFirst` more closely reflected the intended semantics than the previous `break`. I agree its a minor point though.

Reply:
Ok, yes, fair point, I will revert :+1: